x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/TransportPutFollowAction.java

```diff
@@ -124,7 +124,8 @@ private void createFollowerIndex(
         }
         if (leaderIndexMetaData.getSettings().getAsBoolean(IndexSettings.INDEX_SOFT_DELETES_SETTING.getKey(), false) == false) {
             listener.onFailure(
-                new IllegalArgumentException("leader index [" + request.getLeaderIndex() + "] does not have soft deletes enabled"));
+                new IllegalArgumentException("leader index [" + request.getLeaderIndex() + "] does not have soft deletes enabled." +
+                    "soft deletes must be enabled when the index is created by setting index.soft_deletes.enabled to true"));
```

Comment:
Please use `IndexSettings.INDEX_SOFT_DELETES_SETTING.getKey()` so we have fewer places we have to search the code base for the hard string for, if ever we need to.